tests/components/cpuspeed/test_sensor.py

```diff
@@ -61,3 +61,26 @@ async def test_sensor(
     assert state.attributes.get(ATTR_ARCH) == "aargh"
     assert state.attributes.get(ATTR_BRAND) == "Intel Ryzen 7"
     assert state.attributes.get(ATTR_HZ) == 3.6
+
+
+async def test_sensor_partial_info(
+    hass: HomeAssistant,
+    mock_cpuinfo: MagicMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test the CPU Speed sensor missing info."""
+    mock_config_entry.add_to_hass(hass)
+
+    # Pop some info from the mocked CPUSpeed
+    mock_cpuinfo.return_value.pop("brand_raw")
+    mock_cpuinfo.return_value.pop("arch_string_raw")
+
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+    await async_setup_component(hass, "homeassistant", {})
```

Comment:
Why do we set up the homeassistant integration?

Reply:
Because I copy/pasted the test above as a base to start with.
Will remove this line 👍

Reply:
Done!